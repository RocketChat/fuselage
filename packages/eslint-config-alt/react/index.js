module.exports = {
  extends: '../minimal',
  plugins: ['react', 'react-hooks'],
  parser: '@babel/eslint-parser',
  rules: {
    'jsx-quotes': ['error', 'prefer-single'],
    'react/display-name': 'error',
    'react/jsx-uses-react': 'error',
    'react/jsx-uses-vars': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-fragments': ['error', 'syntax'],
    'react/no-multi-comp': 'error',
    'react/react-in-jsx-scope': 'error',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
    'react': {
      version: 'detect',
    },
  },
  env: {
    browser: true,
    es6: true,
  },
  overrides: [
    {
      files: ['**/*.ts', '**/*.tsx'],
      extends: '../typescript',
      parser: '@typescript-eslint/parser',
      plugins: ['react', 'react-hooks'],
      rules: {
        'jsx-quotes': ['error', 'prefer-single'],
        'react/display-name': 'error',
        'react/jsx-uses-react': 'error',
        'react/jsx-uses-vars': 'error',
        'react/jsx-no-undef': 'error',
        'react/jsx-fragments': ['error', 'syntax'],
        'react/no-multi-comp': 'error',
        'react/react-in-jsx-scope': 'error',
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn',
      },
      env: {
        browser: true,
        es6: true,
      },
      settings: {
        'import/resolver': {
          node: {
            extensions: ['.js', '.jsx', '.ts', '.tsx'],
          },
        },
        'react': {
          version: 'detect',
        },
      },
    },
    {
      files: [
        '**/*.stories.js',
        '**/*.stories.jsx',
        '**/*.stories.ts',
        '**/*.stories.tsx',
      ],
      rules: {
        'react/display-name': 'off',
        'react/no-multi-comp': 'off',
      },
    },
  ],
};
