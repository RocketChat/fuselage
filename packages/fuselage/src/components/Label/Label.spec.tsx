import { render } from '@testing-library/react';
import React from 'react';

import { Label } from '.';

it('renders without crashing', () => {
  render(<Label />);
});
