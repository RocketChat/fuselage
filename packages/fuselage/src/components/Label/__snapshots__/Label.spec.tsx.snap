// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Label Component] renders Default without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label"
    >
      Label
    </label>
  </div>
</body>
`;

exports[`[Label Component] renders Disabled without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label--disabled rcx-label"
    >
      Label
    </label>
  </div>
</body>
`;

exports[`[Label Component] renders Required without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label--required rcx-label"
    >
      Label
    </label>
  </div>
</body>
`;
